syntax = "proto3";
// import "google/protobuf/duration.proto";
import "google/protobuf/timestamp.proto";
import "google/protobuf/wrappers.proto";

package indigo;

message DeviceChange {
    google.protobuf.StringValue name = 1;
}

message IndigoEvent {
    google.protobuf.StringValue measurement = 1;
    google.protobuf.Timestamp time = 5;
    IndigoTags tags = 3;
    IndigoFields fields = 4;
}

message IndigoFields {
    google.protobuf.StringValue name = 1;
    google.protobuf.FloatValue lastChanged = 2;
    google.protobuf.FloatValue lastSuccessfulComm = 3;
    google.protobuf.FloatValue id = 4;

    google.protobuf.FloatValue brightness = 10;
    google.protobuf.BoolValue onState = 11;
    google.protobuf.BoolValue state_onOffState = 12;
    google.protobuf.FloatValue onState_num = 13;

    google.protobuf.FloatValue state_humidityInput1 = 21;
    google.protobuf.FloatValue coolSetpoint = 22;
    google.protobuf.BoolValue coolIsOn = 23;
    google.protobuf.FloatValue coolIsOn_num = 24;
    google.protobuf.FloatValue heatSetpoint = 25;
    google.protobuf.BoolValue heatIsOn = 26;
    google.protobuf.FloatValue heatIsOn_num = 27;
    google.protobuf.FloatValue state_temperatureInput1 = 29;
}

message IndigoTags {
    google.protobuf.StringValue name = 1;
    google.protobuf.StringValue folder = 2;
    google.protobuf.StringValue folderId = 3;
}

message HvacFields {
    google.protobuf.FloatValue dehumidifierIsOn_num = 1;
    google.protobuf.FloatValue supportsCoolSetpoint_num = 2;
    google.protobuf.BoolValue state_hvacOperationModeIsOff = 3;
    google.protobuf.BoolValue state_hvacHeaterIsOn = 4;
    google.protobuf.StringValue pluginId = 5;
    google.protobuf.FloatValue lastSuccessfulComm = 6;
    google.protobuf.StringValue protocol = 7;
    google.protobuf.BoolValue state_hvacFanIsOn = 8;
    google.protobuf.StringValue errorState = 9;
    google.protobuf.FloatValue state_hvacDehumidifierIsOn_num = 10;
    google.protobuf.StringValue state_humidityInputsAll = 11;
    google.protobuf.StringValue state_hvacOperationMode_ui = 12;
    google.protobuf.BoolValue state_hvacFanModeIsAlwaysOn = 13;
    google.protobuf.FloatValue hvacMode = 14;
    google.protobuf.FloatValue state_hvacCoolerIsOn_num = 15;
    google.protobuf.FloatValue state_hvacOperationMode_num = 16;
    google.protobuf.FloatValue temperatureSensorCount_num = 17;
    google.protobuf.BoolValue state_hvacOperationModeIsHeat = 18;
    google.protobuf.FloatValue state_setpointHeat = 19;
    google.protobuf.FloatValue state_hvacOperationModeIsOff_num = 20;
    google.protobuf.StringValue name = 21;
    google.protobuf.FloatValue state_temperatureInput1 = 22;
    google.protobuf.FloatValue humidifierIsOn_num = 23;
    google.protobuf.BoolValue state_hvacFanModeIsAuto = 24;
    google.protobuf.FloatValue supportsAllOff_num = 25;
    google.protobuf.FloatValue enabled_num = 26;
    google.protobuf.FloatValue fanIsOn_num = 27;
    google.protobuf.BoolValue state_hvacOperationModeIsProgramHeat = 28;
    google.protobuf.BoolValue configured = 29;
    google.protobuf.FloatValue state_hvacOperationModeIsHeat_num = 30;
    google.protobuf.FloatValue remoteDisplay_num = 31;
    google.protobuf.FloatValue buttonGroupCount_num = 32;
    google.protobuf.FloatValue coolIsOn_num = 33;
    google.protobuf.BoolValue state_hvacOperationModeIsProgramCool = 34;
    google.protobuf.StringValue displayStateValUi = 35;
    google.protobuf.FloatValue hvacMode_num = 36;
    google.protobuf.FloatValue coolSetpoint = 37;
    google.protobuf.BoolValue state_hvacDehumidifierIsOn = 38;
    google.protobuf.BoolValue state_hvacOperationModeIsProgramAuto = 39;
    google.protobuf.FloatValue version = 40;
    google.protobuf.StringValue displayStateId = 41;
    google.protobuf.FloatValue fanMode_num = 42;
    google.protobuf.FloatValue state_hvacFanMode = 43;
    google.protobuf.BoolValue supportsHvacFanMode = 44;
    google.protobuf.FloatValue supportsAllLightsOnOff_num = 45;
    google.protobuf.FloatValue humiditySensorCount_num = 46;
    google.protobuf.StringValue address = 47;
    google.protobuf.FloatValue id_num = 48;
    google.protobuf.BoolValue supportsCoolSetpoint = 49;
    google.protobuf.FloatValue state_hvacFanMode_num = 50;
    google.protobuf.FloatValue lastChanged = 51;
    google.protobuf.FloatValue version_num = 52;
    google.protobuf.StringValue displayStateImageSel = 53;
    google.protobuf.FloatValue state_hvacFanModeIsAuto_num = 54;
    google.protobuf.StringValue displayStateValRaw = 55;
    google.protobuf.BoolValue fanIsOn = 56;
    google.protobuf.FloatValue humiditySensorCount = 57;
    google.protobuf.FloatValue state_humidityInputsAll_num = 58;
    google.protobuf.StringValue state_temperatureInputsAll = 59;
    google.protobuf.StringValue subModel = 60;
    google.protobuf.BoolValue supportsAllOff = 61;
    google.protobuf.FloatValue state_hvacOperationModeIsProgramAuto_num = 62;
    google.protobuf.FloatValue state_hvacOperationMode = 63;
    google.protobuf.FloatValue temperatureSensorCount = 64;
    google.protobuf.BoolValue supportsStatusRequest = 65;
    google.protobuf.FloatValue supportsHeatSetpoint_num = 66;
    google.protobuf.FloatValue supportsHvacFanMode_num = 67;
    google.protobuf.FloatValue state_hvacFanModeIsAlwaysOn_num = 68;
    google.protobuf.FloatValue heatIsOn_num = 69;
    google.protobuf.FloatValue folderId = 70;
    google.protobuf.BoolValue supportsHeatSetpoint = 71;
    google.protobuf.BoolValue humidifierIsOn = 72;
    google.protobuf.FloatValue supportsStatusRequest_num = 73;
    google.protobuf.BoolValue enabled = 74;
    google.protobuf.StringValue state_hvacFanMode_ui = 75;
    google.protobuf.BoolValue remoteDisplay = 76;
    google.protobuf.BoolValue supportsAllLightsOnOff = 77;
    google.protobuf.FloatValue state_hvacHumidifierIsOn_num = 78;
    google.protobuf.FloatValue state_hvacOperationModeIsCool_num = 79;
    google.protobuf.BoolValue state_hvacHumidifierIsOn = 80;
    google.protobuf.FloatValue state_hvacOperationModeIsProgramCool_num = 81;
    google.protobuf.BoolValue heatIsOn = 82;
    google.protobuf.BoolValue state_hvacOperationModeIsCool = 83;
    google.protobuf.FloatValue buttonGroupCount = 84;
    google.protobuf.FloatValue id = 85;
    google.protobuf.BoolValue state_hvacOperationModeIsAuto = 86;
    google.protobuf.FloatValue state_setpointCool = 87;
    google.protobuf.StringValue deviceTypeId = 88;
    google.protobuf.FloatValue configured_num = 89;
    google.protobuf.BoolValue dehumidifierIsOn = 90;
    google.protobuf.FloatValue state_hvacOperationModeIsAuto_num = 91;
    google.protobuf.FloatValue folderId_num = 92;
    google.protobuf.BoolValue supportsHvacOperationMode = 93;
    google.protobuf.FloatValue state_hvacHeaterIsOn_num = 94;
    google.protobuf.StringValue description = 95;
    google.protobuf.FloatValue state_temperatureInputsAll_num = 96;
    google.protobuf.FloatValue supportsHvacOperationMode_num = 97;
    google.protobuf.FloatValue state_hvacOperationModeIsProgramHeat_num = 98;
    google.protobuf.BoolValue coolIsOn = 99;
    google.protobuf.FloatValue heatSetpoint = 100;
    google.protobuf.FloatValue fanMode = 101;
    google.protobuf.BoolValue state_hvacCoolerIsOn = 102;
    google.protobuf.FloatValue state_humidityInput1 = 103;
    google.protobuf.FloatValue state_hvacFanIsOn_num = 104;
    google.protobuf.StringValue model = 105;
}

message BinarySwitchFields {
    google.protobuf.StringValue subModel = 1;
    google.protobuf.StringValue protocol = 2;
    google.protobuf.BoolValue configured = 3;
    google.protobuf.FloatValue buttonConfiguredCount_num = 4;
    google.protobuf.StringValue pluginId = 5;
    google.protobuf.FloatValue buttonGroupCount_num = 6;
    google.protobuf.FloatValue lastSuccessfulComm = 7;
    google.protobuf.FloatValue buttonGroupCount = 8;
    google.protobuf.FloatValue id = 9;
    google.protobuf.FloatValue onState_num = 10;
    google.protobuf.BoolValue supportsAllOff = 11;
    google.protobuf.StringValue errorState = 12;
    google.protobuf.BoolValue remoteDisplay = 13;
    google.protobuf.FloatValue configured_num = 14;
    google.protobuf.BoolValue supportsAllLightsOnOff = 15;
    google.protobuf.StringValue displayStateValUi = 16;
    google.protobuf.FloatValue version = 17;
    google.protobuf.StringValue displayStateId = 18;
    google.protobuf.BoolValue state_onOffState = 19;
    google.protobuf.BoolValue supportsStatusRequest = 20;
    google.protobuf.FloatValue folderId_num = 21;
    google.protobuf.FloatValue remoteDisplay_num = 22;
    google.protobuf.StringValue description = 23;
    google.protobuf.BoolValue onState = 24;
    google.protobuf.FloatValue supportsAllLightsOnOff_num = 25;
    google.protobuf.StringValue address = 26;
    google.protobuf.FloatValue folderId = 27;
    google.protobuf.FloatValue id_num = 28;
    google.protobuf.FloatValue buttonConfiguredCount = 29;
    google.protobuf.StringValue name = 30;
    google.protobuf.FloatValue supportsStatusRequest_num = 31;
    google.protobuf.FloatValue lastChanged = 32;
    google.protobuf.FloatValue state_onOffState_num = 33;
    google.protobuf.FloatValue version_num = 34;
    google.protobuf.BoolValue enabled = 35;
    google.protobuf.FloatValue supportsAllOff_num = 36;
    google.protobuf.StringValue deviceTypeId = 37;
    google.protobuf.FloatValue enabled_num = 38;
    google.protobuf.StringValue displayStateImageSel = 39;
    google.protobuf.StringValue displayStateValRaw = 40;
    google.protobuf.StringValue model = 41;
}

message SecurityFields {
    google.protobuf.StringValue subModel = 1;
    google.protobuf.StringValue protocol = 2;
    google.protobuf.BoolValue state_bypass_nobypass = 3;
    google.protobuf.BoolValue configured = 4;
    google.protobuf.BoolValue state_state_closed = 5;
    google.protobuf.StringValue state_state = 6;
    google.protobuf.FloatValue state_state_tripped_num = 7;
    google.protobuf.StringValue pluginId = 8;
    google.protobuf.FloatValue buttonGroupCount_num = 9;
    google.protobuf.FloatValue lastSuccessfulComm = 10;
    google.protobuf.FloatValue buttonGroupCount = 11;
    google.protobuf.FloatValue id = 12;
    google.protobuf.FloatValue state_bypass_nobypass_num = 13;
    google.protobuf.BoolValue supportsAllOff = 14;
    google.protobuf.StringValue errorState = 15;
    google.protobuf.FloatValue state_state_open_num = 16;
    google.protobuf.BoolValue remoteDisplay = 17;
    google.protobuf.FloatValue configured_num = 18;
    google.protobuf.BoolValue supportsAllLightsOnOff = 19;
    google.protobuf.StringValue displayStateValUi = 20;
    google.protobuf.StringValue displayStateId = 21;
    google.protobuf.FloatValue state_LastChangedTimer = 22;
    google.protobuf.BoolValue state_state_tripped = 23;
    google.protobuf.BoolValue supportsStatusRequest = 24;
    google.protobuf.FloatValue folderId_num = 25;
    google.protobuf.FloatValue remoteDisplay_num = 26;
    google.protobuf.StringValue description = 27;
    google.protobuf.FloatValue state_bypass_bypassed_num = 28;
    google.protobuf.BoolValue state_state_open = 29;
    google.protobuf.StringValue state_LastChangedShort = 30;
    google.protobuf.FloatValue supportsAllLightsOnOff_num = 31;
    google.protobuf.StringValue address = 32;
    google.protobuf.FloatValue folderId = 33;
    google.protobuf.FloatValue id_num = 34;
    google.protobuf.FloatValue state_LastChangedTimer_num = 35;
    google.protobuf.FloatValue state_state_closed_num = 36;
    google.protobuf.StringValue name = 37;
    google.protobuf.FloatValue supportsStatusRequest_num = 38;
    google.protobuf.FloatValue lastChanged = 39;
    google.protobuf.BoolValue state_bypass_bypassed = 40;
    google.protobuf.BoolValue enabled = 41;
    google.protobuf.StringValue state_bypass = 42;
    google.protobuf.FloatValue supportsAllOff_num = 43;
    google.protobuf.StringValue deviceTypeId = 44;
    google.protobuf.FloatValue enabled_num = 45;
    google.protobuf.StringValue displayStateImageSel = 46;
    google.protobuf.StringValue displayStateValRaw = 47;
    google.protobuf.StringValue model = 48;
}

message DimmerSwitchFields {
    google.protobuf.StringValue subModel = 1;
    google.protobuf.FloatValue supportsRGBandWhiteSimultaneously_num = 2;
    google.protobuf.StringValue protocol = 3;
    google.protobuf.FloatValue defaultBrightness_num = 4;
    google.protobuf.BoolValue configured = 5;
    google.protobuf.StringValue address = 6;
    google.protobuf.FloatValue buttonConfiguredCount_num = 7;
    google.protobuf.StringValue pluginId = 8;
    google.protobuf.FloatValue remoteDisplay_num = 9;
    google.protobuf.FloatValue buttonGroupCount_num = 10;
    google.protobuf.FloatValue state_onOffState_num = 11;
    google.protobuf.FloatValue lastSuccessfulComm = 12;
    google.protobuf.FloatValue version_num = 13;
    google.protobuf.FloatValue buttonGroupCount = 14;
    google.protobuf.BoolValue supportsWhiteTemperature = 15;
    google.protobuf.FloatValue supportsTwoWhiteLevels_num = 16;
    google.protobuf.StringValue displayStateValUi = 17;
    google.protobuf.FloatValue onState_num = 18;
    google.protobuf.BoolValue supportsAllOff = 19;
    google.protobuf.StringValue errorState = 20;
    google.protobuf.FloatValue supportsColor_num = 21;
    google.protobuf.BoolValue remoteDisplay = 22;
    google.protobuf.FloatValue configured_num = 23;
    google.protobuf.FloatValue state_brightnessLevel = 24;
    google.protobuf.BoolValue supportsAllLightsOnOff = 25;
    google.protobuf.FloatValue id = 26;
    google.protobuf.BoolValue supportsWhite = 27;
    google.protobuf.FloatValue state_brightnessLevel_num = 28;
    google.protobuf.FloatValue version = 29;
    google.protobuf.StringValue displayStateId = 30;
    google.protobuf.BoolValue state_onOffState = 31;
    google.protobuf.BoolValue supportsColor = 32;
    google.protobuf.FloatValue brightness_num = 33;
    google.protobuf.BoolValue supportsRGBandWhiteSimultaneously = 34;
    google.protobuf.BoolValue onBrightensToLast = 35;
    google.protobuf.StringValue description = 36;
    google.protobuf.FloatValue id_num = 37;
    google.protobuf.BoolValue onState = 38;
    google.protobuf.BoolValue onBrightensToDefaultToggle = 39;
    google.protobuf.FloatValue supportsAllLightsOnOff_num = 40;
    google.protobuf.FloatValue supportsRGB_num = 41;
    google.protobuf.BoolValue supportsRGB = 42;
    google.protobuf.FloatValue defaultBrightness = 43;
    google.protobuf.FloatValue folderId = 44;
    google.protobuf.BoolValue supportsTwoWhiteLevels = 45;
    google.protobuf.FloatValue supportsWhiteTemperature_num = 46;
    google.protobuf.FloatValue supportsWhite_num = 47;
    google.protobuf.FloatValue buttonConfiguredCount = 48;
    google.protobuf.StringValue name = 49;
    google.protobuf.FloatValue supportsStatusRequest_num = 50;
    google.protobuf.FloatValue brightness = 51;
    google.protobuf.FloatValue supportsTwoWhiteLevelsSimultaneously_num = 52;
    google.protobuf.FloatValue onBrightensToDefaultToggle_num = 53;
    google.protobuf.FloatValue lastChanged = 54;
    google.protobuf.BoolValue enabled = 55;
    google.protobuf.FloatValue onBrightensToLast_num = 56;
    google.protobuf.BoolValue supportsTwoWhiteLevelsSimultaneously = 57;
    google.protobuf.FloatValue folderId_num = 58;
    google.protobuf.FloatValue supportsAllOff_num = 59;
    google.protobuf.StringValue deviceTypeId = 60;
    google.protobuf.FloatValue enabled_num = 61;
    google.protobuf.StringValue displayStateImageSel = 62;
    google.protobuf.StringValue displayStateValRaw = 63;
    google.protobuf.BoolValue supportsStatusRequest = 64;
    google.protobuf.StringValue model = 65;
}

message GenericFields {
    google.protobuf.FloatValue lastSuccessfulComm = 1;
    google.protobuf.FloatValue id = 2;
    google.protobuf.StringValue name = 3;
    google.protobuf.FloatValue lastChanged = 4;
}

message IndigoUnknownMessage {
    google.protobuf.StringValue measurement = 1;
    google.protobuf.Timestamp time = 2;
    IndigoTags tags = 4;
    oneof fields {
        BinarySwitchFields binary = 11;
        DimmerSwitchFields dimmer = 12;
        HvacFields hvac = 13;
        SecurityFields security = 14;
        GenericFields generic = 15;
    }
}
